ospfd/ospf_opaque.c

```diff
@@ -746,6 +759,49 @@ int ospf_opaque_is_owned(struct ospf_lsa *lsa)
 	return (oipt != NULL && lookup_opaque_info_by_id(oipt, lsa) != NULL);
 }
 
+/*
+ * Cleanup Link-Local LSAs assocaited with an interface that is being deleted.
+ * Since these LSAs are stored in the area link state database (LSDB) as opposed
+ * to a separate per-interface, they must be deleted from the area database.
+ * Since their flooding scope is solely the deleted OSPF interface, there is no
+ * need to attempt to flush them from the routing domain. For link local LSAs
+ * originated via the OSPF server API, LSA deletion before interface deletion
+ * is required so that the callback can access the OSPF interface address.
+ */
+void ospf_opaque_type9_lsa_if_cleanup(struct ospf_interface *oi)
+{
+	struct route_table *table;
+	struct route_node *rn;
+	struct ospf_lsdb *lsdb;
+	struct ospf_lsa *lsa;
+
+	lsdb = oi->area->lsdb;
+	table = lsdb->type[OSPF_OPAQUE_LINK_LSA].db;
+	for (rn = route_top(table); rn; rn = route_next(rn))
```

Comment:
nit: I see we can reuse our existing pattern `LSDB_LOOP` + `OPAQUE_LINK_LSDB` for this loop/check. It also verifies that the table is not null here.

Reply:
Fixed.